overlord/snapstate/handlers.go

```diff
@@ -1030,6 +1030,7 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) error {
 		// Invoke the hard refresh flow. Upon success the returned lock will be
 		// held to prevent snap-run from advancing until UnlinkSnap, executed
 		// below, completes.
+		// XXX: should we skip it if type is snap.TypeSnapd?
```

Comment:
CC @pedronis - should we ignore snapd here (not create runinhibit lock for it)?; I noticed this with TestDoUnlinkCurrentSnapSnapdNop test which now hits run-inhibit-snap-for-unlink fakeop (only op since it's testing just unlik-current-snap handler). Or do we want to keep it for consistency?

Reply:
if it's easy to, we should probably opt out snapd from the refresh-app-awareness logic, same for core. We don't plan to have gating hooks for it, nor it as any normal apps in it, nor it uses the usual snap data dirs or has the usual security profiles. If it's easy to opt out it would mean we don't risk a refresh-app-awareness bug blocking snapd updates.

Reply:
I'd look at this separately as a followup PR if that's ok?

Reply:
I've proposed a separate PR for this: #11690